File: app/src/main/java/com/jbekas/cocoin/adapter/TodayViewFragmentAdapter.java
package com.jbekas.cocoin.adapter;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.jbekas.cocoin.fragment.TodayViewFragment;
import com.jbekas.cocoin.util.CoCoinUtil;

/**
 * Created by 伟平 on 2015/10/20.
 */

public class TodayViewFragmentAdapter extends FragmentStatePagerAdapter {

    private static int TODAY_VIEW_FRAGMENT_NUMBER = 8;

    public TodayViewFragmentAdapter(FragmentManager fm) {
        super(fm);
    }

    @Override
    public Fragment getItem(int i) {
        return TodayViewFragment.newInstance(i);
    }

    @Override
    public int getCount() {
        return TODAY_VIEW_FRAGMENT_NUMBER;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return CoCoinUtil.GetTodayViewTitle(position % TODAY_VIEW_FRAGMENT_NUMBER);
    }

    @Override
    public int getItemPosition(Object object) {
        return POSITION_NONE;
    }
}

File: app/src/main/kotlin/com/jbekas/cocoin/adapter/TodayViewFragmentAdapter.kt
package com.jbekas.cocoin.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.jbekas.cocoin.fragment.TodayViewFragment

class TodayViewFragmentAdapter(fragmentActivity: FragmentActivity?) : FragmentStateAdapter(
    fragmentActivity!!) {
    override fun createFragment(position: Int): Fragment {
        return TodayViewFragment.newInstance(position)
    }

    override fun getItemCount(): Int {
        return TODAY_VIEW_FRAGMENT_NUMBER
    }

    companion object {
        private const val TODAY_VIEW_FRAGMENT_NUMBER = 8
    }
}